examples/vision/gradient_centralization.py

```diff
@@ -0,0 +1,282 @@
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(32, (3, 3), activation="relu"),
+        layers.Dropout(0.5),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.Dropout(0.5),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.MaxPooling2D(2, 2),
+        layers.Flatten(),
+        layers.Dropout(0.5),
+        layers.Dense(512, activation="relu"),
+        layers.Dense(1, activation="sigmoid"),
+    ]
+)
+
+"""
+We will also create a callback which allows us to easily measure the total training time
+and the time taken for each epoch  since we are interested in comparing the effect of
+Gradient Centralization on the model we built above.
+"""
+
+
+class TimeHistory(tf.keras.callbacks.Callback):
+    def on_train_begin(self, logs={}):
+        self.times = []
+
+    def on_epoch_begin(self, batch, logs={}):
+        self.epoch_time_start = time()
+
+    def on_epoch_end(self, batch, logs={}):
+        self.times.append(time() - self.epoch_time_start)
+
+
+"""
+# Train the model without GC
+"""
+
+"""
+We now train the model we built earlier without Gradient Centralization which we can
+compare to the training performance of the model trained with Gradient Centralization.
+"""
+
+time_callback_no_gc = TimeHistory()
+model.compile(
+    loss="binary_crossentropy",
+    optimizer=RMSprop(learning_rate=1e-4),
+    metrics=["accuracy"],
+)
+
+model.summary()
+
+"""
+We also save the history since we later want to compare our model trained with and not
+trained with Gradient Centralization
+"""
+
+history_no_gc = model.fit(
+    train_ds, epochs=10, verbose=1, callbacks=[time_callback_no_gc]
+)
+
+"""
+# Train the model with GC 
+
+We will now train the same model, this time using Gradient Centralization. We will now
+subclass the `RMSProp` optimizer class modifying the
+`tf.keras.optimizers.Optimizer.get_gradients()` method where we now implement Gradient
+Centralization. On a high level the idea is that let us say we obtain our gradients
+through back propogation for a Dense or Convolution layer we then compute the mean of the
+column vectors of the weight matrix, and then remove the mean from each column vector.
+
+The experiments in [this paper](https://arxiv.org/abs/2004.01461) on various
+applications, including general image classification, fine-grained image classification,
+detection and segmentation and Person ReID demonstrate that GC can consistently improve
+the performance of DNN learning.
+"""
+
+
+class gc_rmsprop(RMSprop):
+    def get_gradients(self, loss, params):
+        # We here just provide a modified get_gradients() function since we are
+        # trying to just compute the centralized gradients.
+
+        grads = []
+        for grad in K.gradients(loss, params):
```

Comment:
Sure thing, also since we are making a subclass of `tf.keras.optimizer.RMSProp` and only care about the `get_gradients()` method as you mentioned earlier I have used `super().get_gradients()` to make it even more concise.